### DISCUSSION TITLE
Support for Fleet API?

### DISCUSSION THREAD

**Discussion by adampetrovic** [teslamate-org/teslamate#3865] [Ideas]
For those of us running the new Fleet API proxy server, is there any appetite for allowing Teslamate to direct requests to the locally running proxy instead of hitting the owners API?

**Comment by brianmay** [comment]
See https://github.com/teslamate-org/teslamate/discussions/3792

Summary:

There are at least two people making proxy servers available, which would ease the pains in setting this up, configuring public/private keys, etc. https://teslemetry.com/ is one.  https://app.myteslamate.com/ is another.

The downside is that the streaming API - particularly the streaming API Teslamate uses - is not part of the Fleet API. Teslamate would probably have to switch to polling model, and no doubt we would be seriously rate limited.

It might be possible to continue using this streaming API and the new fleet API at the same time, but the docs specifically say "Per the Fleet API policies, continued usage of unsupported APIs will result in the revocation of Fleet API access privileges."

The new Fleet Telemetry API - for streaming - does not support older MCU1 or MCU2 cars at this point in time.

**Reply by bmerciergallay** [reply]
what is the rate limit of the telemetry api ?
maybe the lack of older MCU support is a bummer but if the streaming api gets shut, that is still the only way to retain most of teslamate features. (or even any functioning feature as well)

If teslamate finally switches to fleet apis, the suggestion above is good although it would be good to keep the option as well to setting up its own proxy for people with the skill necessary to still have a free option.

**Reply by brianmay** [reply]
Rate limit currently "Is limited to 20 fields total, and 10 values per minute. (This should increase in the future)". I suspect this might be restrictive compared with what teslamate currently logs... 20 fields total might not be enough (???)

https://teslemetry.com/docs/getting-started/telemetry#limitations

The fact that MCU2 is not supported means I cannot contribute in any way, because I can't even test it. And I am not likely to want to purchase another Tesla again anytime in the foreseeable future. Even if I needed a new EV, would look at the competition first I think, not happy with the direction Tesla is going. If somebody wanted to donate me a MCU3 car though... :-)

A lot of the existing teslamate code involves polling, calculating the next poll time, reconnecting to the legacy streaming websocket, etc. All of these would become redundant I think. Hence it might be simpler to create a new project from scratch that receives telemetry events, calculates start/stop drives/charges, and then writes this to the existing database scheme and publishes to the existing MQTT topics.

**Reply by KrzysztofHajdamowicz** [reply]
There is another FleetAPI proxy, from Tessie:
https://dash.tessie.com/integrations/fleet-api
